ng-deploy/src/builders/actions/deploy.ts

```diff
@@ -1,8 +1,40 @@
-import { execSync, spawnSync } from "child_process";
+import { BuilderContext } from '@angular-devkit/architect/src/index2';
+import { virtualFs, experimental, normalize } from '@angular-devkit/core';
+import { Stats } from '@angular-devkit/core/src/virtual-fs/host';
+import { CoreSchemaRegistry, transforms } from '@angular-devkit/core/src/json/schema';
+import { join } from 'path';
+import { FirebaseTools } from '../../ng-deploy/types';
 
-export default function deploy() {
-    const bin = execSync('(cd ' + __dirname + ' && npm bin)').toString().trim();
-    const firebaseBin = bin + '/firebase';
-    spawnSync(firebaseBin, ['login'], {stdio: 'inherit'});
-    spawnSync(firebaseBin, ['deploy', '--only', 'hosting'], {stdio: 'inherit'});
+export default async function deploy(firebaseTools: FirebaseTools, context: BuilderContext, host: virtualFs.Host<Stats>,) {
+  try {
+    await firebaseTools.list();
+  } catch (e) {
+    context.logger.warn("🚨 You're not logged into Firebase. Logging you in...");
```

Comment:
Should we check for error type maybe? Are there any other cases where it could throw?

Reply:
Yes, good point. I'll look at the error shape.